app/assets/stylesheets/content/_tooltips.sass

```diff
@@ -26,27 +26,160 @@
 // See doc/COPYRIGHT.rdoc for more details.
 //++
 
-/***** Tooltips *****
+//***** Tooltips *****
 
-.tooltip
+// tooltips used for arbitrary elements
+
+$tooltipBackground: #e3f5ff
+$tooltipBorder: none
+$tooltipHeight: 22px
+$tooltipTextColor: $body-font-color
+$tooltipArrowSize: 6px
+
+%top-bottom
+  &:before, &:after
+    left: 50%
+    transform: translateX(-50%)
+
+  &:after
+    width: auto
+
+  &.-multiline
+    &:before, &:after
+      transform: translateY(50%)
+      margin-bottom: 0
+      filter: blur(10px)
+
+%left-right
+  &:before, &:after
+    bottom: 50%
+  &:before
+    margin-bottom: $tooltipArrowSize * -1 + 1
+  &:after
+    margin-bottom: $tooltipHeight/-1.5
+
+  &.-multiline
+    &:before, &:after
+      transform: translateY(50%)
+      margin-bottom: 0
+
+@mixin colorize($color:$tooltipBackground, $textColor:$tooltipTextColor)
+  @each $position in top, right, bottom, left
+    &.tooltip-#{$position}
+      &:before
+        border-#{$position}-color: $color
+      &:after
+        background-color: $tooltipBackground
+        color: $tooltipTextColor
+
+[data-tooltip]
   position: relative
-  z-index: 24
-  &.hover, &:hover
-    z-index: 25
-    color: #000
-  span.tip
-    display: none
-    text-align: left
-
-div.tooltip
-  &:hover span.tip, &.hover span.tip
-    display: block
+  display: inline-block
+  box-sizing: content-box
+  // TODO: this "fixes" the spacing issue in text, we need to find a better way for that
+  padding-right: 5px
+
+  &:before, &:after
     position: absolute
-    top: 12px
-    left: 24px
-    width: 270px
-    border: 1px solid #555
-    background-color: #fff
-    padding: 4px
-    font-size: 0.8em
-    color: #505050
+    visibility: hidden
+    opacity: 0
+    z-index: 99999
+    box-sizing: content-box
+    transform: translate3d(0, 0, 0)
+
+  &:before
+    content: ''
+    border: $tooltipArrowSize solid transparent
+
+  &:after
+    height: $tooltipHeight
+    padding: $tooltipHeight/2 $tooltipHeight/2 0 $tooltipHeight/2
+    font-size: 13px
+    line-height: $tooltipHeight/2
+    white-space: nowrap
+    content: attr(data-tooltip)
```

Comment:
@myabc nope, that's why i added the aria- attribute in https://github.com/opf/openproject/pull/2795/files#diff-11ffa4e8cc3c0e094b49260b53095ea8R9